lib/task/digitalobject/digitalObjectDeleteTask.class.php

```diff
@@ -50,71 +53,132 @@ protected function configure()
 
     $t = new QubitTimer;
-    $id = QubitPdo::fetchColumn('SELECT object_id FROM slug WHERE slug=?', array($arguments['slug']));
 
-    if (!$id)
+    // Remind user they are in dry run mode
+    if ($options['dry-run'])
+    {
+      $this->logSection('digital-object', '*** DRY RUN (no changes will be made to the database) ***');
+    }
+
+    if ($options['media-type'] && !array_key_exists($options['media-type'], $this->validMediaTypes))
+    {
+      error_log(sprintf('Invalid value for "media-type", must be one of (%s)',
+        implode(',', array_keys($this->validMediaTypes))));
+
+      exit(1);
+    }
+
+    $sql = "SELECT slug.object_id, object.class_name \r
+              FROM slug \r
+              JOIN object ON object.id = slug.object_id \r
+              WHERE slug.slug = '" . $arguments['slug'] . "'";
+
+    $statement = QubitPdo::prepareAndExecute($sql);
+
+    $row = $statement->fetch(PDO::FETCH_ASSOC);
+
+    if (!$row['object_id'])
     {
       throw new sfException('Invalid slug "'.$arguments['slug'].'" entered.');
     }
 
-    if (null === $infoObj = QubitInformationObject::getById($id))
+    if (!in_array($row['class_name'], array('QubitInformationObject', 'QubitRepository')))
     {
-      throw new sfException('Failed to fetch information object with the slug given.');
+      throw new sfException('Invalid slug with object type "'.$row['class_name'].'" entered.');
     }
 
-    $ios = $options['and-descendants'] ? $infoObj->descendants->andSelf() : array($infoObj);
-    $nDeleted = 0;
+    if ($row['class_name'] == 'QubitInformationObject' &&
+      null === $informationObject = QubitInformationObject::getById($row['object_id']))
+    {
+      throw new sfException('Failed to fetch information object with the slug given.');
+    }
+    else if ($row['class_name'] == 'QubitRepository' &&
+      null === $repository = QubitRepository::getById($row['object_id']))
+    {
+      throw new sfException('Failed to fetch repository with the slug given.');
+    }
 
-    foreach ($ios as $io)
+    switch ($row['class_name'])
     {
-      $this->logSection('digital-object', sprintf('(%d of %d) deleting digital object for: %s',
-                        ++$nDeleted, count($ios), $io->getTitle(array('cultureFallback' => true))));
+      case 'QubitInformationObject':
+        $objectIds = $options['and-descendants'] ? $this->getInformationObjectChildren($informationObject) : array($informationObject->id);
+
+        break;
+
+      case 'QubitRepository':
+        // Get all linked top level information object recs.
+        $relatedInformationObjects = $repository->getRelatedInformationObjectIds();
+
+        foreach ($relatedInformationObjects as $id)
+        {
+          $object = QubitInformationObject::getById($id);
+
+          // Always include descendants when deleting by repository.
+          $objectIds = array_merge($objectIds, $this->getInformationObjectChildren($object));
+        }
```

Comment:
I'd try to avoid the ORM in here to amalgamate the `$objectIds` ...

Instead of reusing `$repository->getRelatedInformationObjectIds()`, you could write a new query to get the `rgt` and `lft` values instead of the `id` and use them to call `getInformationObjectChildren`.

Reply:
Good idea. I have made changes to the queries to remove reliance on the ORM info obj.